### ISSUE TITLE
unnecessary_override should allow tightened return type

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#627]
the following case shouldn't lint

```dart
class A {
  Object m1() => null;
}
class B extends A {
  @override int m1() => super.m1(); // OK
}

**Comment by a14n** [comment]
A workaround that seems to work is:

```dart
class B extends A {
  @override int m1();
}
```

**Comment by a14n** [comment]
However the semantic is not exactly the same. In checked mode a type check is done for the first snippet but not for the workaround snippet.

**Comment by Hixie** [comment]
The workaround snippet is bogus (the B.m1 is entirely ignored, I think). We probably need a lint that prevents that from happening at all.
See also https://github.com/dart-lang/sdk/issues/28250.

**Comment by natebosch** [comment]
Fixed in https://github.com/dart-lang/linter/pull/748
